probe/kubernetes/client.go

```diff
@@ -151,34 +135,50 @@ func NewClient(config ClientConfig) (Client, error) {
 
 	podStore := NewEventStore(result.triggerPodWatches, cache.MetaNamespaceKeyFunc)
 	result.podStore = result.setupStore(c.CoreV1Client.RESTClient(), "pods", &apiv1.Pod{}, podStore)
-
 	result.serviceStore = result.setupStore(c.CoreV1Client.RESTClient(), "services", &apiv1.Service{}, nil)
 	result.nodeStore = result.setupStore(c.CoreV1Client.RESTClient(), "nodes", &apiv1.Node{}, nil)
 	result.namespaceStore = result.setupStore(c.CoreV1Client.RESTClient(), "namespaces", &apiv1.Namespace{}, nil)
+	result.deploymentStore = result.setupStore(c.ExtensionsV1beta1Client.RESTClient(), "deployments", &apiextensionsv1beta1.Deployment{}, nil)
+	result.daemonSetStore = result.setupStore(c.ExtensionsV1beta1Client.RESTClient(), "daemonsets", &apiextensionsv1beta1.DaemonSet{}, nil)
+	result.jobStore = result.setupStore(c.BatchV1Client.RESTClient(), "jobs", &apibatchv1.Job{}, nil)
+	result.cronJobStore = result.setupStore(c.BatchV2alpha1Client.RESTClient(), "cronjobs", &apibatchv2alpha1.CronJob{}, nil)
+	result.statefulSetStore = result.setupStore(c.AppsV1beta1Client.RESTClient(), "statefulsets", &apiappsv1beta1.StatefulSet{}, nil)
 
-	// We list deployments here to check if this version of kubernetes is >= 1.2.
-	// We would use NegotiateVersion, but Kubernetes 1.1 "supports"
-	// extensions/v1beta1, but not deployments or daemonsets.
-	if _, err := c.Extensions().Deployments(metav1.NamespaceAll).List(metav1.ListOptions{}); err != nil {
-		log.Infof("Deployments and DaemonSets are not supported by this Kubernetes version: %v", err)
-	} else {
-		result.deploymentStore = result.setupStore(c.ExtensionsV1beta1Client.RESTClient(), "deployments", &apiextensionsv1beta1.Deployment{}, nil)
-		result.daemonSetStore = result.setupStore(c.ExtensionsV1beta1Client.RESTClient(), "daemonsets", &apiextensionsv1beta1.DaemonSet{}, nil)
+	return result, nil
+}
+
+func (c *client) isResourceSupported(resource string) (bool, error) {
+	var group string
+	switch resource {
+	case "deployments", "daemonsets":
+		group = "extensions/v1beta1"
+	case "cronjobs":
+		group = "batch/v2alpha1"
+	case "statefulsets":
+		group = "apps/v1beta1"
+	case "pods", "services", "nodes", "namespaces":
+		group = "v1"
+	case "jobs":
+		group = "batch/v1"
+	default:
+		return false, fmt.Errorf("Support check not available for resource %v", resource)
```

Comment:
We shouldn't need this mapping. The `RESTClient()` passed to `setupStore()` is a `rest.Interface`, which has an `APIVersion()` method that returns a `schema.GroupVersion`. So changing the type on `setupStore()` from `cache.Getter` to `rest.Interface` should give us access to the info we need.